client/app/utils/i18n.js

```diff
@@ -0,0 +1,39 @@
+// This file has two tasks:
+//
+// 1. Load the language bundle if we are in node environment
+// 2. Require the i18n-js library and export it
+//
+// The language bundle is not loaded if we are in the browser
+// environment because there we get the bundle from a separate
+// long cached script-tag
+//
+// The language bundle is loaded in a global I18n variable like this:
+//
+// `global.I18n.translations["en"] = <translation json>`
+//
+// The i18n-js library is able to read the language bundle from that global
+//
+
+import { bind } from 'lodash';
+
+if (typeof window === 'undefined') {
+  try {
+    require('../../../app/assets/javascripts/i18n/all.js'); // eslint-disable-line no-undef
+  } catch (e) {
+    console.warn("Can't load language bundle all.js"); // eslint-disable-line no-console
+  }
+}
+
+// It seems that we need to use `require` here instead of `import`.
+// With `import` the translations seem not to be populated.
+const I18n = require('i18n-js'); // eslint-disable-line no-undef
```

Comment:
Ok! That explains the difference between `import` and `require`. Thanks!

Good questions about the order. I made some changes to the code and try to simplify this a bit.

Actually, the order doesn't matter anymore. But what *does* matter is that the global `I18n` needs to be initialized before loading the `i18n-js` library. See this line in the `i18n-js` code: https://github.com/fnando/i18n-js/blob/2ca6d31365bb41db21e373d126cac00d38d15144/app/assets/javascripts/i18n.js#L31

If the global `I18n` is not initialized before loading the library, the library will create a new object and the translations get loaded to another object.

I haven't tried loading the page with throttled connection. Should I?